src/test/integration/users/AVS.t.sol

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "forge-std/Test.sol";
+
+import "src/contracts/core/AllocationManager.sol";
+import "src/contracts/strategies/StrategyFactory.sol";
+
+import "src/test/mocks/ERC20Mock.sol";
+import "src/test/integration/users/User.t.sol";
+import "src/test/integration/TimeMachine.t.sol";
+import "src/test/utils/Logger.t.sol";
+
+import "src/test/utils/SingleItemArrayLib.sol";
+
+interface IAVSDeployer {
+    function allocationManager() external view returns (AllocationManager);
+    function strategyFactory() external view returns (StrategyFactory);
+    function timeMachine() external view returns (TimeMachine);
+}
+
+contract AVS is Logger, IAllocationManagerTypes {
+    using SingleItemArrayLib for *;
+
+    Vm constant cheats = Vm(VM_ADDRESS);
+    AllocationManager immutable allocationManager;
+    StrategyFactory immutable strategyFactory;
+    TimeMachine immutable timeMachine;
+    string _NAME;
+
+    constructor(
+        string memory name
+    ) {
+        IAVSDeployer deployer = IAVSDeployer(msg.sender);
+        allocationManager = deployer.allocationManager();
+        strategyFactory = deployer.strategyFactory();
+        timeMachine = deployer.timeMachine();
+        _NAME = name;
+    }
+
+    modifier createSnapshot() virtual {
+        timeMachine.createSnapshot();
+        _;
+    }
+
+    receive() external payable {}
+
+    function NAME() public view override returns (string memory) {
+        return _NAME;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// AllocationManager Interactions
+    /// -----------------------------------------------------------------------
+    
+    function createOperatorSets(
+        uint256 numOpSets,
+        uint256 numStrategies
+    ) public createSnapshot returns (uint32[] memory operatorSetIds, IStrategy[][] memory strategies) {
+        _logM("createOperatorSets");
+
+        CreateSetParams[] memory p = new CreateSetParams[](numOpSets);
+        operatorSetIds = new uint32[](numOpSets);
+        strategies = new IStrategy[][](numOpSets);
+
+        for (uint256 i; i < numOpSets; ++i) {
+            p[i].operatorSetId = uint32(cheats.randomUint({bits: 32}));
+            for (uint256 j; j < numStrategies; ++j) {
+                IERC20 token = IERC20(new ERC20Mock());
+                p[i].strategies[j] = strategyFactory.deployNewStrategy(token);
```

Comment:
This function will create operatorSets for the passed in opSets & strategies. The strategies should be those that are deployed in the integration deployer. In general, we should leave randomization to be outside of the user contracts